base/cvd/cuttlefish/host/commands/cvd/parser/instance/cf_connectivity_configs.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <fmt/format.h>
+#include <json/json.h>
+
+#include "common/libs/utils/result.h"
+
+namespace cuttlefish {
+
+std::string GenerateGroupFlag(const Json::Value& instances_json) {
+  std::vector<std::string> vsock_groups;
+
+  for (const auto& instance_json : instances_json) {
+    vsock_groups.emplace_back(
+        instance_json["connectivity"]["vsock"]["guest_group"].asString());
```

Comment:
Will this abort if 'vsock' or 'guest_group' are not in the config file?

Reply:
I suspected the same - to address did some manual testing with valid and blank values, assigned to arbitrary VMs in the cluster.  The result is actually desirable here, I get a blank return value in case of unspecified, which also is valid in context of the group argument expects (blank value when not used.).